lib/synapse/service_watcher/zookeeper.rb

```diff
@@ -54,7 +59,7 @@ def discover
         node = @zk.get("#{@discovery['path']}/#{id}")
 
         begin
-          host, port, name = deserialize_service_instance(node.first)
+          host, port, name, weight = deserialize_service_instance(node.first)
```

Comment:
Yea, we're working on exposing this to HAProxy. This is the first step (see https://github.com/Yelp/synapse/pull/14 for discussion between the author and me regarding this)